app_test.go

```diff
@@ -196,6 +196,68 @@ func TestAppStart(t *testing.T) {
 		require.Error(t, err, "expected start failure")
 		assert.Contains(t, err.Error(), "can't invoke non-function")
 	})
+
+	t.Run("ProvideProvide", func(t *testing.T) {
```

Comment:
I'll try to come up with the better name but that's literally what is being verified: fx.Providing an fx.Provide.